components/org.wso2.carbon.identity.pat/org.wso2.carbon.identity.pat.core.service/src/main/java/org/wso2/carbon/identity/pat/core/service/grant/PATGrantHandler.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com). All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content.
+ */
+
+package org.wso2.carbon.identity.pat.core.service.grant;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.base.IdentityRuntimeException;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenRespDTO;
+import org.wso2.carbon.identity.oauth2.model.RequestParameter;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.handlers.grant.AbstractAuthorizationGrantHandler;
+import org.wso2.carbon.identity.pat.core.service.common.PATConstants;
+import org.wso2.carbon.identity.pat.core.service.common.PATUtil;
+import org.wso2.carbon.identity.pat.core.service.dao.PATDAOFactory;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
+import org.wso2.carbon.user.core.common.User;
+import org.wso2.carbon.user.core.service.RealmService;
+
+public class PATGrantHandler extends AbstractAuthorizationGrantHandler {
+
+    private static final Log log = LogFactory.getLog(PATGrantHandler.class);
+
+    @Override
+    public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+
+        RequestParameter[] parameters = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getRequestParameters();
+
+        String validityPeriod = getValueFromRequestParameters(parameters, PATConstants.VALIDITY_PERIOD);
+        if (StringUtils.isNotBlank(validityPeriod)) {
+            tokReqMsgCtx.setValidityPeriod(Long.parseLong(validityPeriod));
+        }
+
+        OAuth2AccessTokenRespDTO responseDTO = super.issue(tokReqMsgCtx);
+
+        String alias = getValueFromRequestParameters(parameters, PATConstants.ALIAS);
+        String description = getValueFromRequestParameters(parameters, PATConstants.DESCRIPTION);
+
+        if (StringUtils.isNotBlank(alias) && StringUtils.isNotBlank(description)) {
```

Comment:
handle these validation inside the API core service before calling OAuth2 services